src/libraries/icubclient/src/subsystems/subSystem_emotion.cpp

```diff
@@ -0,0 +1,69 @@
+#include <yarp/os/all.h>
+#include <algorithm>
+#include "icubclient/subsystems/subSystem_emotion.h"
+
+bool icubclient::SubSystem_emotion::connect() {
+    if(yarp::os::Network::isConnected(portRPC.getName(), "/icub/face/emotions/in")) {
+        return true;
+    } else {
+        return yarp::os::Network::connect(portRPC.getName(), "/icub/face/emotions/in");
+    }
+}
+
+icubclient::SubSystem_emotion::SubSystem_emotion(const std::string &masterName) : SubSystem(masterName) {
+    portRPC.open(("/" + m_masterName + "/emotion:rpc").c_str());
+    m_type = SUBSYSTEM_EMOTION;
+}
+
+void icubclient::SubSystem_emotion::Close() {
+    portRPC.interrupt();
+    portRPC.close();
+}
+
+bool icubclient::SubSystem_emotion::setEmotion(Emotion emotion) {
+    return setEmotion(emotion, Part::all);
+}
+
+bool icubclient::SubSystem_emotion::setEmotion(Emotion emotion, Part part) {
+    yarp::os::Bottle bReq, bResp;
+    std::string spart, semotion;
+
+    switch(emotion)
```

Comment:
This is really ugly. Why don't you create a `std::pair` which contains the pairs?